### ISSUE TITLE
Some displays not recognized anymore after update

### ISSUE THREAD

**Issue by Nilstrieb** [NixOS/nixpkgs#299944]
### Describe the bug
Only one of my three displays is recognized and displays anything. `xrandr` only showed that one display. The non-working displays were connected with display port, the working one was HDMI.

This happened after updating (I am on unstable): https://github.com/Nilstrieb/nixos/commit/3039fb6454aeae6730d5d1fabc895cc25dbb1fdd

I will attempt to bisect the regression.

### Steps To Reproduce
I don't really have concise steps to reproduce, it happened when upgrading from 20f77aa09916374aa3141cbc605c955626762c9a to 2726f127c15a4cc9810843b96cad73c7eb39e443. I use X11 with GNOME.

### Expected behavior
All 3 displays are working.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
# This is the working version, as I can't properly show the non-working one.
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.22, NixOS, 24.05 (Uakari), 24.05.20240321.20f77aa`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - channels(nils): `"nixos"`
 - nixpkgs: `/nix/store/wa8lhchql76i6czzvad3bqz6lvnjnbr5-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Nilstrieb** [comment]
I think the regression was in 1ed5df1e703135d2d6814a761ce67e6e8073af30, I will try to verify that.

**Comment by Nilstrieb** [comment]
Yes, it was that commit. @K900 do you have an idea what could be causing this problem? I know nothing about graphics, so I'm happy to fix my config of there was a problem somewhere.

**Comment by Nilstrieb** [comment]
CPU: AMD Ryzen 9 3950X, GPU: NVIDIA GeForce RTX 2080 Ti

**Comment by K900** [comment]
Are you running the proprietary Nvidia driver or Nouveau?

**Comment by Nilstrieb** [comment]
I think I am using the proprietary driver: https://github.com/Nilstrieb/nixos/blob/03be64e31a82968c42f2de5fae6787e971ca798c/nixos/configuration.nix#L127

**Comment by K900** [comment]
Can you try setting `hardware.nvidia.modesetting.enable = true;`?

**Comment by shimunn** [comment]
Same propblem here on an ThinkPad T470s via Intel HD Graphics

**Comment by librephoenix** [comment]
I'm having this same issue on an Asus Rog Strix G15. I'm using Hyprland.

CPU: AMD Ryzen 9 5900HX
GPU: AMD Radeon RX 6800M

Also this if it's helpful:
```
[emmet@snowfire:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
- host os: `Linux 6.8.0-zen1, NixOS, 24.05 (Uakari), 24.05.20240321.20f77aa`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.18.2`
- channels(root): `"nixos-22.11"`
- channels(emmet): `"home-manager, nixpkgs"`
- nixpkgs: `/home/emmet/.nix-defexpr/channels/nixpkgs`
```

**Comment by jackschu** [comment]
I get an issue where xrandr only picks up a `None-1 connected` after updating from 23.11 to 24.05 where it used to pick up `DP-3 connected primary` and so i get terrible resolution.

trying `hardware.nvidia.modesetting.enable = true;` xrandr does pick up the display seemingly correctly but i get some very weird graphical hiccups (flashing, failing to select, transparency not working)

For now going to stay on 23.11 and see how this shakes out.

---
My info
I'm running with a `NVIDIA GeForce RTX 2070 SUPER` and `AMD Ryzen 5 3600X 6-Core Processor`
with the proprietary drivers

Heres `nix-shell -p nix-info --run "nix-info -m"`
```console
[jackschu@nixos:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.37, NixOS, 24.05 (Uakari), 24.05.2539.49ee0e94463a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by carlthome** [comment]
Same issue for me on a 2080 Ti and 5600X.

```sh
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.43, NixOS, 24.05 (Uakari), 24.05.20240730.cf05eea`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(carl): `""`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/z3vrsz16ypv60m2pp0gi7x7gpx40gfh1-source`
```

After `hardware.nvidia.modesetting.enable = true;` resolution was set for my display but lots of graphical glitches.
